File: src/main/java/dev/todaka/robustredis/protocol/RedisCommand.java
package dev.todaka.robustredis.protocol;

import io.netty.buffer.ByteBuf;

public class RedisCommand<R> {
    public final CommandInput commandInput;
    public final CommandOutput<R> commandOutput;

    public RedisCommand(CommandInput commandInput, CommandOutput<R> commandOutput) {
        this.commandInput = commandInput;
        this.commandOutput = commandOutput;
    }

    public void writeToByteBuf(ByteBuf buf) {
        commandInput.writeToByteBuf(buf);
    }

    public String getFirstKey() {
        return commandInput.getKeys().get(0);
    }
}

File: src/main/kotlin/dev/todaka/robustredis/model/RedisCommand.kt
package dev.todaka.robustredis.model

import io.netty.buffer.ByteBuf

data class RedisCommand<R>(
    val commandInput: CommandInput,
    val commandOutput: CommandOutput<R>
) {
    fun writeToByteBuf(buf: ByteBuf) {
        commandInput.writeToByteBuf(buf)
    }
}
